src/python/pants/backend/docker/target_types.py

```diff
@@ -662,6 +662,13 @@ def is_applicable(cls, target: Target) -> bool:
         """Whether to provide additional tags for this target or not."""
         return True
 
+    @classmethod
+    def image_tags_rule(cls):
```

Comment:
A single base class could have multiple such classmethods, for different families of rules, if it made sense.